### ISSUE TITLE
5.2.0 doesn't install dependencies correctly

### ISSUE THREAD

**Issue by isnifer** [npm/npm#17755]
## TL;DR
npm install deps of third-party package into **%package_name%/node_modules** instead of **node_modules**

#### I'm opening this issue because:
  - [x] npm is producing an incorrect install.

#### What's going wrong?
npm doesn't install dependencies correctly. That's why I can't use eslint for example.

#### How can the CLI team reproduce the problem?
1. Init new project
2. Add two deps into project
```json
"eslint": "^3.13.1",
"eslint-config-tipsi": "1.5.0"
```
3. Run `npm i`

### supporting information:

 - `npm -v` prints: 5.2.0
 - `node -v` prints: v8.1.2
 - `npm config get registry` prints: https://registry.npmjs.org/
 - Windows, OS X/macOS, or Linux?: macOS

### I've created gif to demonstrate issue, sorry for 3 minutes, but it is the comparison with yarn:
![npm_doesnot_install_deps](https://user-images.githubusercontent.com/1788245/28154788-33a65490-67b4-11e7-8fd2-3819582a6e0c.gif)

**Comment by LitoMore** [comment]
Related to #17717

**Comment by ctrlplusb** [comment]
npm@5.2.0
node@8.1.2
https://registry.npmjs.org/
macOS@10.12.5

---

I am experiencing the same issue.

I have a project with two dependencies (`foo` and `bar`), both depending on the exact version of `babel-loader@7.1.1`.  After running the install for my project `babel-loader` exists at:

```
/node_modules/foo/node_modules/babel-loader
/node_modules/bar/node_modules/babel-loader
```

Note the nesting within `foo` and `bar`, even though they are both depending on the _exactly_ same version of `babel-loader`.

I did a grep to see if any of my other dependencies were also depending on `babel-loader`. I found one (`cssnano`), but it lists `babel-loader@6.4.1` within its `devDependencies`. This shouldn't have an impact in the flattening algorithm right?  I would assume that the `devDependencies` belonging to each of my projects dependencies would be ignored.

I did a recursive search across my node_modules and could find no other `babel-loader` installs.  So the nested structure is even more confusing.

Also, to be clear, no other packages are depending on `babel-loader`, even via `peerDependencies`.

---

I have created a new git repo to help illustrate this issue:
https://github.com/ctrlplusb/npm-issue-17755

I also created two new npm packages both just depending on `babel-loader@7.1.1`:

   - ctrlplusb-npm-issue-foo
   - ctrlplusb-npm-issue-bar

My repo depends on these, as well as `cssnano` (which has `babel-loader` in its `devDependencies`).

Steps to reproduce:
```
git clone https://github.com/ctrlplusb/npm-issue-17755
cd npm-issue-17755
npm install 
find ./node_modules | grep -i babel-loader
```

**Comment by doasync** [comment]
Related to this fix (npm 5.2): #17678

> This initial fix is naive, never flattening peer deps, and we can look into doing something more sophisticated later on. (@iarna)

It is reverted and is going to be patched soon...

**Comment by doasync** [comment]
@isnifer, if you want to fix it fast, I suggest to replace:

`"eslint": "^3"`
to
`"eslint-config-airbnb-standard": "^1"`

[This](https://www.npmjs.com/package/eslint-config-airbnb-standard) is my eslint-on-steroids)

**Comment by isnifer** [comment]
@doasync I've installed deps via yarn - it helps

**Comment by ctrlplusb** [comment]
npm@5.3.0 resolves my issue.  `babel-loader` is now at the expected location 👍

**Comment by isnifer** [comment]
@ctrlplusb fine. I will try to upgrade)

**Comment by isnifer** [comment]
Fixed in **5.3.0**
